packages/nextjs/components/Header.tsx

```diff
@@ -78,18 +78,23 @@ export const Header = () => {
   const isLocalNetwork = targetNetwork.id === devnet.id;
 
   const { provider } = useProvider();
-  const { address, status } = useAccount();
+  const { address, status, chainId } = useAccount();
   const [isDeployed, setIsDeployed] = useState(true);
 
   useEffect(() => {
-    if (status === "connected" && address) {
-      provider.getContractVersion(address).catch((e) => {
-        if (e.toString().includes("Contract not found")) {
-          setIsDeployed(false);
-        }
-      });
+    if (status === "connected" && address && chainId === targetNetwork.id) {
+      provider
+        .getContractVersion(address)
```

Comment:
lets change this to use `getClassHashAt` , i experienced some weird behavior using this helper function. When moving from non deployed account to another non deployed account, also this will be similar to the way we do it [here](https://github.com/Quantum3-Labs/scaffold-stark-2/blob/33111d7ab6deb94e359c30d3aa917402e0dba865/packages/nextjs/hooks/scaffold-stark/useDeployedContractInfo.ts#L41) in `useScaffoldContractInfo()`

Reply:
cc : @Nadai2010